### ISSUE TITLE
PDF export function no longer working

### ISSUE THREAD

**Issue by spadgezilla** [atlassian-api/atlassian-python-api#1158]
Hi, I'm having an issue with the PDF export function, this was working up until May 1st, I wonder if recent changes found here could be to blame? https://confluence.atlassian.com/cloud/blog/2023/05/atlassian-cloud-changes-apr-24-to-may-1-2023

I'm using confluence.export_page(page_id) and getting 'PDF conversion not successful' in return. It also seems to return the html content of the page in IDLE shell so it's doing something, I can't remember if it used to do this though.

Is anyone else having this problem?

**Comment by rosomri** [comment]
Same

**Comment by rosomri** [comment]
@torrid @spadgezilla 
Hi there! 
I've just created a PR that aims to solve this issue.
It seems to be working fine for me, would appreciate your feedback.

**Here's the current flow works:**
1.  Create a pdf page export task using the "spaces/flyingpdf/pdfpageexport.action?pageId={pageId}" endpoint.
2. In the next step, use the poll endpoint, "runningtaskxml.action?taskId=[taskId]", to determine whether the export task has been completed.
3. Retrieve the download URL from the poll endpoint response if the task was successfully completed
4. Get the PDF data from the retrieved endpoint.

**Step 2 seems to be where the issue appears:**
- It does not matter if the task is successfully initialised or not, the poll endpoint will report that the task does not exist without raising an error message.
- As a result, step 3 queries the space URL without a specific page.
- Consequently, the response is a general HTML page without any error messages.

**The problems are as follows:**
- There seems to be a problem with the poll endpoint. Atlassian recently updated Confluence's PDF export functionality.
- An error should be thrown if the task ID is invalid.

**Optional solution:**
**TLDR;**
**The functionality is working great, but the PR has not yet been completed.**

- Atlassian recently updated Confluence's PDF export functionality. To get the export task's current state, `task/{taskId}/progress` endpoint should be used instead of the deprecated `runningtaskxml.action?taskId={taskId}`. 
- Also the flow of retrieving the content from the download signed URL was changed a bit.
In order to download a PDF file from a signed S3 URL, you do not need to provide any authentication or headers. If you do provide those, the request will fail. As the session has already been initiated with those headers, it cannot be used.

**Feel free to edit and elaborate**

**Comment by pp-wdiebel** [comment]
I am running into this error as well, and this appears to be what is going on in the python:

![image](https://github.com/atlassian-api/atlassian-python-api/assets/78034628/86512ba9-9ada-4021-9b54-90622fa2346a)

Here we can see that it gets the response and attempts to decode it in UTF as shown here:

![image](https://github.com/atlassian-api/atlassian-python-api/assets/78034628/6d622074-cb2f-4681-a9f6-49389b9f1898)

I also added in the print statement so we can get a better idea of what the response looks like now:

![image](https://github.com/atlassian-api/atlassian-python-api/assets/78034628/09ee30c1-1670-49e7-87c2-783f8185ed67)

I would agree that something has changed in how the API is responding now which likely is due to this may release note:

![image](https://github.com/atlassian-api/atlassian-python-api/assets/78034628/60a898bc-ad50-4952-9123-53bb46a8d604)

I will keep tinkering around but I am seeing does not at all appear to be a download link like is expected but is actual PDF content.  Likely there needs to be some fundamental updates to address this new functionality

**Comment by pp-wdiebel** [comment]
After more digging I have been able to develop a fix for the new changes.  As noted above the behavior of the API has changed to now provide a direct PDF back instead of a download link.

By adding this function to get the PDF instead of the URL:
![image](https://github.com/atlassian-api/atlassian-python-api/assets/78034628/ed99244d-548d-4315-aea5-f2e14214a00c)

We can now use this to call instead of the prior get url in the export page
![image](https://github.com/atlassian-api/atlassian-python-api/assets/78034628/3272e4f5-102b-4d04-969b-3b67fb5eb70f)

here i am now returning it directly to PDF and then returning that PDF.  That may not be best practice someone else will have to answer for that.  

However now when export is called it will produce a PDF that can be written directly as shown below:
![image](https://github.com/atlassian-api/atlassian-python-api/assets/78034628/3f83830e-b166-42e2-9894-66c5a472a5b5)

Here you can see that I call the export, get a pdf file then write it to an actual file.  Here is a sample of the code and the file written.

![image](https://github.com/atlassian-api/atlassian-python-api/assets/78034628/3abdc563-177a-4191-92c8-224f617ca7f3)

![image](https://github.com/atlassian-api/atlassian-python-api/assets/78034628/1ebb789e-0645-4b02-a7ea-43832024e5f8)


I am also uploading a copy of the python confluence package i updated.  This will "work" but likely needs to be better absorbed by the team.  @rosomri  i added your changes.

@torrid @spadgezilla  the file i have provided below should resolve the issue for now

**Comment by pp-wdiebel** [comment]
something else to note is that it does appear that there is a 75 second timeout on the export to PDF requests that causes issues.

For pages where I experience this via API I am noticing they fail to export to PDF via the front end as well so this seems to be a problem with the PDF exporter that tangentially hits this.

**Comment by torrid** [comment]
Bugfix #1161 works for me, thanksalot. The Atlassian rendering got much slower though, I had a timeout routine that never before caught action, now 2 files in 50...

**Comment by pp-wdiebel** [comment]
> Bugfix #1161 works for me, thanksalot. The Atlassian rendering got much slower though, I had a timeout routine that never before caught action, now 2 files in 50...

I have noticed on the pages that time-out for me the actual time out value doesn't mater (i pumped it up to 5 minutes from the base 75 and nobody else got through) and all of the pages that fail on the timeout ALSO fail through the front end PDF export.

I suspect this is something with the shared PDF export as I have noticed the URL that the python calls is the same you find coming out of the front end exporter
